### PR TITLE
MAKE-1248: logging and duplicate key handling

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: queue/remote_ordered.go [hunk 1]
```diff
@@ -96,33 +83,42 @@ func (q *remoteSimpleOrdered) Next(ctx context.Context) amboy.Job {
...
+					"job":       id,
+					"edges":     edges,
+					"dep":       dep.Type(),
+					"job_type":  job.Type().Name,
+					"num_edges": len(edges),
+					"message":   "job is blocked",
+				})
+
 				if len(edges) == 1 {
+					// this is just an optimization; if there's one dependency its easy
```
Comment (Kimchelly) [inline queue/remote_ordered.go:101]: it's

File: queue/remote_ordered.go [hunk 2]
```diff
@@ -96,33 +83,42 @@ func (q *remoteSimpleOrdered) Next(ctx context.Context) amboy.Job {
...
+					// to move that job *up* in the queue by submitting it here. there's a
```
Comment (Kimchelly) [inline queue/remote_ordered.go:102]: Remove asterisks? Putting formatting in a comment seems weird. Also capitalize for full sentences.

File: queue/remote_ordered.go [hunk 3]
```diff
@@ -96,33 +83,42 @@ func (q *remoteSimpleOrdered) Next(ctx context.Context) amboy.Job {
...
+					// chance, however, that it's already in progress and we'll end up
+					// running it twice.
```
Comment (Kimchelly) [inline queue/remote_ordered.go:104]: Is it fine to run twice?

Comment (tychoish) [reply queue/remote_ordered.go:104]: this is just copied. I think there's a risk that we run things twice, but it's not certain.

**Pull request merged**